dev/devicelab/bin/run.dart

```diff
@@ -66,12 +68,40 @@ Future<void> main(List<String> rawArgs) async {
       localEngineSrcPath: localEngineSrcPath,
     );
 
-    if (!result['success'])
-      exitCode = 1;
-
     print('Task result:');
     print(const JsonEncoder.withIndent('  ').convert(result));
     section('Finished task "$taskName"');
+
+    if (!result['success']) {
+      exitCode = 1;
+      if (args['exit']) {
+        return;
+      }
+    }
+  }
+}
+
+void addTasks({
+  List<ManifestTask> tasks,
+  ArgResults args,
+  List<String> taskNames,
+}) {
+  if (args.wasParsed('continue-from')) {
+    final int index = tasks.indexWhere((ManifestTask task) => task.name == args['continue-from']);
+    if (index == -1) {
+      throw 'Invalid task name "${args['continue-from']}"';
+    }
+    tasks.removeRange(0, index);
+  }
+  // Only start skipping if user specified a task to continue from
+  bool isQualifyingStage;
```

Comment:
nit: prefer defaulting these to a value instead of null.

Reply:
Actually, if you don't use these outside of the loop I would move them inside

Reply:
Done.